specification/servicefabric/resource-manager/Microsoft.ServiceFabric/2017-07-01-preview/servicefabric.json

```diff
@@ -0,0 +1,3268 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ServiceFabricManagementClient",
+    "description": "Azure Service Fabric Resource Provider API Client",
+    "version": "2017-07-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "flow": "implicit",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      },
+      "type": "oauth2"
+    }
+  },
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/clusters/{clusterName}": {
+      "put": {
+        "operationId": "Clusters_Create",
+        "summary": "Create a ServiceFabric cluster",
+        "description": "Create cluster resource\n",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterBodyParam"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          }
+        ],
+        "tags": [
+          "Cluster"
+        ],
+        "x-ms-examples": {
+          "Put a cluster with min parameter": {
+            "$ref": "./examples/ClusterPutOperation_example_min.json"
+          },
+          "Put a cluster with max parameter": {
+            "$ref": "./examples/ClusterPutOperation_example_max.json"
+          }
+        },
+        "x-ms-long-running-operation": "true",
+        "responses": {

```

Comment:
This is existing API i think, but 201 + provisioningState + AzureAsyncOperation header is recommended for PUT  LRO instead of 202.

Reply:
Cluster doesn't support async operations yet. This will be in another API version.

Reply:
Lets mark "x-ms-long-running-operation": "true" as false then. It is currently set to true. And we can also remove the 202 from the response.